Bazel 7.0 - can't compile  with `/DUNICODE` `/D_UNICODE`

### Description of the bug:


We have defaults set to build using the Unicode version of the Microsoft APIs, using `/DUNICODE` `/D_UNICODE` to use the unicode version of the APIs by default. This has broken building `external/bazel_tools/src/main/cpp/util/path_windows.cc` Which should be using the APIs post fixed with "A" where the char type is always narrow.


```ERROR: C:/dev/_bazel/pnebtqjr/external/bazel_tools/src/main/cpp/util/BUILD:34:11: Compiling src/main/cpp/util/path_windows.cc failed: (Exit 2): cl.exe failed: error executing CppCompile command (from target @bazel_tools//src/main/cpp/util:filesystem) C:\bazeltools\vs16\vc\Tools\MSVC\14.29.30133\bin\HostX64\x64\cl.exe @bazel-out/x64_windows-fastbuild/bin/external/bazel_tools/src/main/cpp/util/_objs/filesystem/path_windows.obj.params
external/bazel_tools/src/main/cpp/util/path_windows.cc(99): error C2664: 'DWORD ExpandEnvironmentStringsW(LPCWSTR,LPWSTR,DWORD)': cannot convert argument 1 from 'const _Elem *' to 'LPCWSTR'
        with
        [
            _Elem=char
        ]
external/bazel_tools/src/main/cpp/util/path_windows.cc(99): note: Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or function-style cast
C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\um\processenv.h(198): note: see declaration of 'ExpandEnvironmentStringsW'
external/bazel_tools/src/main/cpp/util/path_windows.cc(108): error C2664: 'DWORD ExpandEnvironmentStringsW(LPCWSTR,LPWSTR,DWORD)': cannot convert argument 1 from 'const _Elem *' to 'LPCWSTR'
        with
        [
            _Elem=char
        ]
external/bazel_tools/src/main/cpp/util/path_windows.cc(108): note: Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or function-style cast
C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\um\processenv.h(198): note: see declaration of 'ExpandEnvironmentStringsW'
```

### Which category does this issue belong to?


C++/Objective-C Rules

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.
Create an empty python binary and build that target
```
py_binary(
    name = "foo",
    srcs = [":foo.py"],
    python_version = "PY3",
    visibility = ["//visibility:public"],
)
```

With an empty windows project run this command. 
`bazel build --copt=/D_UNICODE --copt=/DUNICODE //:foo`

### Which operating system are you running Bazel on?


windows

### What is the output of `bazel info release`?


release 7.0.0rc1

### If `bazel info release` returns `development version` or `(@non-git)`, tell us how you built Bazel.


_No response_

### What's the output of `git remote get-url origin; git rev-parse master; git rev-parse HEAD` ?


_No response_

### Is this a regression? If yes, please try to identify the Bazel commit where the bug was introduced.


Yes, moving to bazel 7.0.0rc1

### Have you found anything relevant by searching the web?

_No response_

### Any other information, logs, or outputs that you want to share?

```ERROR: C:/dev/_bazel/pnebtqjr/external/bazel_tools/src/main/cpp/util/BUILD:34:11: Compiling src/main/cpp/util/path_windows.cc failed: (Exit 2): cl.exe failed: error executing CppCompile command (from target @bazel_tools//src/main/cpp/util:filesystem) C:\bazeltools\vs16\vc\Tools\MSVC\14.29.30133\bin\HostX64\x64\cl.exe @bazel-out/x64_windows-fastbuild/bin/external/bazel_tools/src/main/cpp/util/_objs/filesystem/path_windows.obj.params
external/bazel_tools/src/main/cpp/util/path_windows.cc(99): error C2664: 'DWORD ExpandEnvironmentStringsW(LPCWSTR,LPWSTR,DWORD)': cannot convert argument 1 from 'const _Elem *' to 'LPCWSTR'
        with
        [
            _Elem=char
        ]
external/bazel_tools/src/main/cpp/util/path_windows.cc(99): note: Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or function-style cast
C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\um\processenv.h(198): note: see declaration of 'ExpandEnvironmentStringsW'
external/bazel_tools/src/main/cpp/util/path_windows.cc(108): error C2664: 'DWORD ExpandEnvironmentStringsW(LPCWSTR,LPWSTR,DWORD)': cannot convert argument 1 from 'const _Elem *' to 'LPCWSTR'
        with
        [
            _Elem=char
        ]
external/bazel_tools/src/main/cpp/util/path_windows.cc(108): note: Types pointed to are unrelated; conversion requires reinterpret_cast, C-style cast or function-style cast
C:\Program Files (x86)\Windows Kits\10\include\10.0.22621.0\um\processenv.h(198): note: see declaration of 'ExpandEnvironmentStringsW'
```

--- a/src/create_embedded_tools.py
+++ b/src/create_embedded_tools.py
@@ -48,6 +48,7 @@ output_paths = [
     ('*tools/cpp/runfiles/generated_*',
      lambda x: 'tools/cpp/runfiles/' + os.path.basename(x)[len('generated_'):]),
     ('*launcher.exe', lambda x: 'tools/launcher/launcher.exe'),
+    ('*launcher_maker.exe', lambda x: 'tools/launcher/launcher_maker.exe'),
     ('*def_parser.exe', lambda x: 'tools/def_parser/def_parser.exe'),
     ('*zipper.exe', lambda x: 'tools/zip/zipper/zipper.exe'),
     ('*zipper', lambda x: 'tools/zip/zipper/zipper'),

--- a/tools/launcher/BUILD.tools
+++ b/tools/launcher/BUILD.tools
@@ -20,5 +20,18 @@ filegroup(
 
 filegroup(
     name = "launcher_maker",
-    srcs = ["//src/tools/launcher:launcher_maker"],
+    srcs = select({
+        "//src/conditions:host_windows": [":launcher_maker_windows"],
+        "//conditions:default": [
+            "//src/tools/launcher:launcher_maker",
+        ],
+    }),
+)
+
+filegroup(
+    name = "launcher_maker_windows",
+    srcs = select({
+        "//src/conditions:remote": ["//src/tools/launcher:launcher_maker"],
+        "//conditions:default": ["launcher_maker.exe"],
+    }),
 )

